### PR TITLE
Update BungeeCord and Paper to 1.21 and update Gradle Wrapper to 8.9

### PR DESCRIPTION
**Pull request by Obydux**

### CODE REVIEW

**Comment by booky10** [comment]
is there anything wrong with #874, or did you just overlook that pr?

**Comment by Bram1903** [comment]
> is there anything wrong with #874, or did you just overlook that pr?

Besides the point that this probably won't even work, because he didn't fix any sort of backwards compatibility nor the mixin injector for Fabric.

**Comment by Obydux** [comment]
> is there anything wrong with #874, or did you just overlook that pr?

Yup, didn't notice that PR and didn't realize there were more problems with the Fabric update


> > is there anything wrong with #874, or did you just overlook that pr?
> 
> Besides the point that this probably won't even work, because he didn't fix any sort of backwards compatibility. Let alone the entire issue I am having. He's also making a big change of changing to googler's shadow fork which he doesn't mention nor discuss anywhere. And then we have a bunch of new repos which I am not sure what they exactly add.

Switching to gooler's shadow fork is required to be able to build the Fabric mod with Java 21, also not sure what do you mean by "bunch" of repos, I only added a new repo for brigadier 1.2.9 which is required for BungeeCord 1.21 API and the other one was for MockBukkit 1.21

**Comment by Obydux** [comment]
> > Switching to googler's shadow fork is required to be able to build the Fabric mod with Java 21, also not sure what do you mean by "bunch" of repos, I only added a new repo for brigadier 1.2.9 which is required for BungeeCord 1.21 API and the other one was for MockBukkit 1.21
> 
> ### 1. Googler's Shadow Fork
> 
> I understand that using Googler's Shadow Fork is necessary, as the latest version of Fabric Loom requires Java 21. This is why I proposed the change to switch to Googler's fork in my PR, instead of simply switching it without mentioning it anywhere. Documenting such changes is crucial. But even more important, testing the changes on the latest version. It seems unconventional to create a PR without verifying the changes, especially with a library like this.
> ### 2. The Repositories
> 
> You are adding repositories for [Jitpack](https://github.com/retrooper/packetevents/pull/893/files#diff-94ffd2c65a13014e00a66162041a62b546cd1d28e1a24b8165f2e0c415f7a357R17) and [Glaremasters](https://github.com/retrooper/packetevents/pull/893/files#diff-ec439b1e2afd65b177a08ec3c49e4485baba5ee049ec71d74017960cd18a5a98R7). I am unclear on the necessity of the Brigadier repository, but I might be overlooking something.
> 
> Another thing I noticed is that you updated the compatibility level of Java in the Fabric module. We are still compiling back to Java 17, and by increasing that value to Java 21, I am pretty sure you are breaking backward compatibility even further? Even though we need certain libraries that require Java 21, that doesn't mean we still can't compile back to Java 17 for Fabric specifically, as with our other modules we even compile it back to Java 8.
> 
> Chances are I am overlooking something with the Brigadier repository, so sorry in advance if I made a mistake there, but I honestly don't understand its requirement.

That version of Brigadier is not available in the central maven repository but is required by BungeeCord since 1.20-R0.2 thus why the Glaremasters repo was added. Jitpack was added to the API because the 1.21 version of MockBukkit is not on their Maven yet.

My mistake for not explaining why I did these changes, and I agree that it was unnecessary to up the Java compatibility level. Should I close the PR or revert some changes?

**Comment by Bram1903** [comment]
> My mistake for not explaining why I did these changes, and I agree that it was unnecessary to up the Java compatibility level. Should I close the PR or revert some changes?

Thanks for your explanation. I am still a bit confused about the brigadier stuff because as far as I know we aren't using that at all. I would wait for a response from Booky or another maintainer, because perhaps some of the changes could benefit us, but just in a standalone PR, or in this PR with a lot of stuff reverted (and some PR description 😄). They will probably explain that to you 👍🏻

File: bungeecord/build.gradle.kts
Comment (booky10) [inline bungeecord/build.gradle.kts:7]: https://repo.papermc.io/service/rest/repository/browse/maven-public/com/mojang/brigadier/1.2.9/

brigadier v1.2.9 also exists in papers maven repo, why are you adding an additional repository for it?

File: testlibs.versions.toml
Comment (booky10) [inline testlibs.versions.toml:4]: im not really sure if we should update to unstable builds of junit if the old ones work just fine

File: libs.versions.toml
Comment (booky10) [inline libs.versions.toml:5]: why not choose viaversion v5.0.1, as its not a snapshot version?

File: api/build.gradle.kts
Comment (booky10) [inline api/build.gradle.kts:16]: if its only for mockbukkit, limit its content then:
```suggestion
    maven("https://jitpack.io/") {
        content {
            includeGroup("com.github.MockBukkit")
        }
    }
```

**Review by booky10 — APPROVED** [review]
other than from my review comments, im also not really sure what this pr is for? does this pr fix any issues?
packetevents works on both bungeecord 1.21 and paper 1.21, if im not mistaken

**Review by booky10 — CHANGES_REQUESTED** [review]
pressed the wrong button, not approved yet

**Comment by retrooper** [comment]
I don't see the purpose either

**Comment by retrooper** [comment]
For now I'll just close this then?

**Pull request closed without merging**
